tests/contract-playground/src/VoidConstructor.sol

```diff
@@ -0,0 +1,36 @@
+// SPDX-License-Identifier: MIT
+pragma solidity ^0.8.0;
+
+contract VoidConstructor {}
+
+contract FilledConstructor {
+    constructor() {
+        /* some stuff */    
+    }
+}
+
+// This contracts think VoidConstructor has a defined constructor and it tries to call it
+contract B is VoidConstructor, FilledConstructor {
+    // BAD invocation (1 instance)
+    constructor() 
+    VoidConstructor() 
+    FilledConstructor() 
+    {
+        
+    }
+}
+
+// BAD invocation (1 instance)
+// NOTE - BUT WE CANNOT CATCH IT (because AST shows no difference between this and the below contract)
+contract C is VoidConstructor() {
```

Comment:
These bottom two appear exactly the same in the AST? Hmmm That sucks

Reply:
The reason i didn't include that test is because, if you are going to pass in parameters, then that couldve only compiled if the constructor was defined

Reply:
Default constructors are the only ones that can be invoked without defining them .